app/models/message.rb

```diff
@@ -168,109 +99,48 @@ def disable_group_email_link(to_person)
     "#{Setting.get(:url, :site)}groups/#{group.id}/memberships/#{to_person.id}?code=#{to_person.feed_code}&email=off"
   end
 
-  def email_from(to_person)
-    if group
-      from_address("#{person.name} [#{group.name}]")
-    else
-      from_address(person.name)
-    end
+  def email_from(*)
+    from_address(group ? "#{person.name} [#{group.name}]" : person.name)
   end
 
   def email_reply_to(to_person)
-    if not to_person.messages_enabled?
-      "\"#{I18n.t('messages.do_not_reply')}\" <#{Site.current.noreply_email}>"
-    else
-      from_address(person.name, :real)
-    end
+    return from_address(person.name, :real) if to_person.messages_enabled?
+    "\"#{I18n.t('messages.do_not_reply')}\" <#{Site.current.noreply_email}>"
   end
 
-  def from_address(name, real=false)
+  def from_address(name, real = false)
     if person.email.present?
-      %("#{name.gsub(/"/, '')}" <#{real ? person.email : Site.current.noreply_email}>)
+      %("#{name.delete('"')}" <#{real ? person.email : Site.current.noreply_email}>)
     else
       "\"#{I18n.t('messages.do_not_reply')}\" <#{Site.current.noreply_email}>"
     end
   end
 
-  before_create :generate_security_code
-
-  def generate_security_code
-    begin
-      code = rand(999999)
-      write_attribute :code, code
-    end until code > 0
-  end
```

Comment:
self.code ||= rand(999_999) +1
Its inside a block for before_save.  I figured instead of running pick a number between 0 and 999999 until it wasnt zero why not add one?